maven-xml/src/test/java/org/apache/maven/xml/sax/LexicalHandlerVerifier.java

```diff
@@ -0,0 +1,277 @@
+import org.apache.maven.xml.sax.filter.AbstractSAXFilter;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+import org.xml.sax.XMLFilter;
+import org.xml.sax.XMLReader;
+import org.xml.sax.ext.LexicalHandler;
+import org.xml.sax.helpers.XMLFilterImpl;
+
+/**
+ * Some tests to help understand the chain of events regarding XMLFilters, XMLReaders, LexicalHandlers and Transformers
+ * 
+ * @author Robert Scholte
+ *
+ */
+public class LexicalHandlerVerifier
+{
+    private static final String SAX_PROPERTIES_LEXICAL_HANDLER = "http://xml.org/sax/properties/lexical-handler";
+    
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Test
+    public void parseXmlReader() throws Exception
+    {
+        expectedException.expect( UnsupportedOperationException.class );
+        expectedException.expectMessage( "LexicalHandlerVerifier" );
+
+        XMLReader reader = Factories.newXMLReader();
+        reader.setProperty( SAX_PROPERTIES_LEXICAL_HANDLER, new UnsupportedOperationExceptionLexicalHandler() );
+
+        InputSource inputSource = new InputSource( new StringReader( "<root><!-- COMMENT --></root>" ) ); 
+        reader.parse( inputSource );
+    }
+
+    @Test
+    public void parseXmlFilter() throws Exception
+    {
+        expectedException.expect( UnsupportedOperationException.class );
+        expectedException.expectMessage( "LexicalHandlerVerifier" );
+
+        XMLReader reader = Factories.newXMLReader();
+        reader.setProperty( SAX_PROPERTIES_LEXICAL_HANDLER, new UnsupportedOperationExceptionLexicalHandler() );
+        
+        XMLFilter filter = new XMLFilterImpl( reader );
+
+        InputSource inputSource = new InputSource( new StringReader( "<root><!-- COMMENT --></root>" ) ); 
+        filter.parse( inputSource );
+    }
+    
+    @Test
+    public void transformXmlReader() throws Exception
+    {
+        Writer writer = new StringWriter();
+        StreamResult result = new StreamResult( writer );
+
+        SAXTransformerFactory transformerFactory = (SAXTransformerFactory) Factories.newTransformerFactory();
+        TransformerHandler transformerHandler = transformerFactory.newTransformerHandler();
+        transformerHandler.setResult( result );
+        Transformer transformer = transformerFactory.newTransformer();
+        transformer.setOutputProperty( OutputKeys.OMIT_XML_DECLARATION, "yes" );
+        
+        Source xmlSource = new StreamSource( new StringReader( "<root><!--COMMENT--></root>" ) );
+        
+        SAXResult transformResult = new SAXResult( transformerHandler );
+        transformResult.setLexicalHandler( new SortCommentLexicalHandler( transformerHandler ) );
+        transformer.transform( xmlSource, transformResult );
+        
+        assertThat( writer.toString() ).and( "<?xml version=\"1.0\" encoding=\"UTF-8\"?>"
```

Comment:
Are you certain that this will work on non-UTF-8 encodings?

Reply:
Yes, `MavenITmng2254PomEncodingTest` in maven-integration-test that is specifically written for encoding issues. 
Without github.com/apache/maven/pull/286/files#diff-9037bac1b0c7ac509f2f375f84d60f8dR1191-R1217 these would fail.